### PR TITLE
Feature - update search request filters

### LINKED ISSUE #3112: Update search Request page with new filter fields
<sup>ℹ️ [Figma (root)][figma] | [Sitemap][sitemap]</sup>
<sup>_This initial comment is collaborative and open to modification by all._</sup>

## Description
The easiest way to handle this may be to allow the SearchRequestFilters component to receive _either_ a PoolCandidateFilter _or_ an ApplicantFilter. If given an ApplicantFilter, then it must display a Skills filter summary and a Employment Duration filter summary, and forgo CmoAssets.

This applies to the Admin's summary view when viewing a request as well.

## Screenshot: Prototype
[![a screenshot of ...][image]][image-link]
<sup>[Click screenshot to visit Figma.][image-link]</sup>

## Acceptance Criteria
- [ ] Page is unchanged if viewing a request with a PoolCandidateFilter
- [ ] If showing an ApplicantFIlter, Skills and Duration filters are shown, and CmoAssets are not
- [ ] The above is true for the search Requests page, *and* for the admin Request page.

   [image]: https://user-images.githubusercontent.com/4313717/174323056-026118db-490f-4b6e-bc87-5bdf49bd6512.png
   [image-link]: https://www.figma.com/file/XS4Ag6GWcgdq2dBlLzBkay/Admin-v3-(Profile-release)?node-id=1164%3A5745
   [figma]: https://go.talent.c4nada.ca/figma
   [sitemap]: https://go.talent.c4nada.ca/sitemap

### PR DESCRIPTION
**Pull request by yonikid15**
Resolves #3112 

# Notes
- The [Request Page](http://localhost:8000/en/talent/request) should be unchanged if viewing a request with a `PoolCandidateFilter`. Once ApplicantSearch feature is released the ApplicantFilters should be shown instead. To test that the `PoolCandidateFilter` still works, you'll need to set the `FEATURE_APPLICANTSEARCH` [flag](https://github.com/GCTC-NTGC/gc-digital-talent/blob/535b05a2b385ce2a3d21dbe48878c8bdbf881ca3/frontend/.apache_env#L12) to `false` and **rebuild** the PHP container.
- When reviewing make sure the changes apply to the view on the admin side as well by viewing a Search Request (eg. `http://localhost:8000/en/admin/search-requests/{uuid}/edit`)

### CODE REVIEW

File: frontend/common/src/components/SearchRequestFilters/SearchRequestFilters.tsx [hunk 1]
```diff
@@ -193,7 +431,6 @@ const SearchRequestFilters: React.FunctionComponent<
     },
   );
   const typeOfOpportunity = ""; // TODO: Replace with data fetched from api
```
Comment (petertgiles) [inline frontend/common/src/components/SearchRequestFilters/SearchRequestFilters.tsx:433]: 'typeOfOpportunity' is assigned a value but never used.eslint@typescript-eslint/no-unused-vars

**Comment by vd1992** [comment]
there is a little something to still get rid of

image

**Comment by vd1992** [comment]
missing data from the applicant filter, arrow pointing to the applicant filter probably not needed in hindsight

image

created request below

image

**Comment by vd1992** [comment]
candidate search is using PoolCandidateFilter and returning everyone (empty query)

image

**Review by vd1992 — CHANGES_REQUESTED** [review]
some bugs to iron out above
yeah, this looks like a finicky issue with the two filters

**Review by petertgiles — COMMENTED** [review]
Looks like The Eagle was already here. :grin:  I'll just add a small linting problem that came up for me.

**Comment by esizer** [comment]
If you merge `main`, the Chromatic Storybook workflow should start working again.

**Comment by yonikid15** [comment]
> If you merge `main`, the Chromatic Storybook workflow should start working again.

Thanks Eric!

File: frontend/common/src/components/SearchRequestFilters/SearchRequestFilters.tsx (hunk 1 above)
Comment (yonikid15) [reply frontend/common/src/components/SearchRequestFilters/SearchRequestFilters.tsx:433]: Oh ya I believe this got deprecated. Thanks 👍

**Comment by yonikid15** [comment]
> some bugs to iron out above yeah, this looks like a finicky issue with the two filters

Thanks for the review @vd1992 Should have fix up soon

**Review by tristan-orourke — APPROVED** [review]
I just wanted to review the overall structure of how data is passed from the Search page to the Request page, and it looks good to me! I left a suggestion in SearchRequestFilters, but its nothing critical.
I'll leave final approval to Vachan, who seems to have done a thorough testing job.

**Comment by vd1992** [comment]
the SHOULD HAVE message is gone
request page fills in info from applicant filter
the searchPoolCandidates still isn't quite resolved yet

image

it is correctly seeing that applicant filter is defined and pool candidate filter is null though

image

File: frontend/common/src/components/SearchRequestFilters/SearchRequestFilters.tsx [hunk 2]
```diff
@@ -45,61 +46,302 @@ const FilterBlock: React.FunctionComponent<FilterBlockProps> = ({
...
-              : intl.formatMessage({
-                  defaultMessage: "N/A",
-                  description: "Text shown when the filter was not selected",
-                })}
-          </p>
-        )}
+          {children}
+        </>
+      ) : (
+        <>
```
[the remark this answers is not in the document]
Comment (yonikid15) [reply frontend/common/src/components/SearchRequestFilters/SearchRequestFilters.tsx]: I came up with a solution without needing to create another component. In FilterBlock component, `content` can be set to undefined, and also `children` will be joined with the two versions of the title formatting.

**Comment by yonikid15** [comment]
> you've requested a review, so lets go
> 
> the SHOULD HAVE message is gone request page fills in info from applicant filter the searchPoolCandidates still isn't quite resolved yet

@vd1992 Yea, so I talked to Tristan about this yesterday and we agreed that it's out of the scope of this issue as it's not related to the `<SearchRequestFilter />` view, so were going to separate that into a new issue. Thanks for the review!

**Review by vd1992 — APPROVED** [review]
Great work! This was a bit of a tricky issue!
Please remember to spin off the issue

**Pull request merged**